### PR TITLE
Restructure k8s documentation

### PR DESCRIPTION
**Pull request by pat-s**
As suggested in https://github.com/woodpecker-ci/woodpecker/discussions/2160.

- Simplified wording
- Referenced helm chart
- Removed "experimental" banner as the k8s backend is quite stable I'd say (after using it for months now)
- Add example for `resources` definition`

### CODE REVIEW

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 1]
```diff
@@ -1,51 +1,72 @@
 # Kubernetes backend
 
-:::caution
-Kubernetes support is still experimental and not all pipeline features are fully supported yet.
+The kubernetes backend executes steps inside standalone pods. A temporary PVC is created for the lifetime of the pipeline to transfer files between steps.
```
Comment (6543) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:3]: I would still call it experimentall or clearly point out, that services dont worl properly jet!

Comment (6543) [reply docs/docs/30-administration/22-backends/40-kubernetes.md:3]: And link to https://github.com/woodpecker-ci/woodpecker/issues/1513

Comment (pat-s) [reply docs/docs/30-administration/22-backends/40-kubernetes.md:3]: Hmm in my perception "experimental" is a different state than what the k8s backend is in. 

I am aware of #1513 and happy to link it but I think it's out of the "experimental" state and in a state in which some features are still missing or could be improved but I would label it at least "beta"?

Comment (6543) [reply docs/docs/30-administration/22-backends/40-kubernetes.md:3]: Well lets call it beta untill we got it 100% feature compatible with the docker one

Comment (6543) [reply docs/docs/30-administration/22-backends/40-kubernetes.md:3]: Well I personnaly would be fine with just clearly mention its current missing feature clearly

Comment (pat-s) [reply docs/docs/30-administration/22-backends/40-kubernetes.md:8]: Sufficient now?

**Comment by pat-s** [comment]
FYI I've also included workflow filter changes (testwise) - happy to remove them again if more discussion is needed. See #2200

File: docs/docs/30-administration/00-setup.md [hunk 2]
```diff
@@ -171,6 +171,16 @@ services:
 +     - WOODPECKER_AGENT_SECRET=${WOODPECKER_AGENT_SECRET}
 ```
 
+### Kubernetes 
+
+We recommended to deploy Woodpecker using the [Woodpecker helm chart](https://github.com/woodpecker-ci/helm).
+Have a look at the `values.yaml` config files for all available settings.
```
Comment (anbraten) [inline docs/docs/30-administration/00-setup.md:177]: We could link the values file here

File: .woodpecker/securityscan.yml [hunk 3]
```diff
@@ -4,6 +4,13 @@ when:
     branch:
       - ${CI_REPO_DEFAULT_BRANCH}
       - release/*
+    path: &when_path
+      - "agent/**"
+      - "cli/**"
+      - "cmd/**"
+      - "server/**"
+      - "shared/**"
+      - "woodpecker-go/**"
```
[the remark this answers is not in the document]
Comment (pat-s) [reply .woodpecker/securityscan.yml:13]: Fair enough - even though I'd argue that running in it CRON once a day would be sufficient?

File: docs/docs/30-administration/00-setup.md (hunk 2 above)
Comment (pat-s) [reply docs/docs/30-administration/00-setup.md:177]: I'd prefer the top level link - as the sentence is before the reference to `values.yml`.

**Comment by pat-s** [comment]
@qwerty287 Outsourced the WF modifications to #2203

File: .woodpecker/securityscan.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (pat-s) [reply .woodpecker/securityscan.yml]: Ah interesting, I didn't check for this. It's only a few seconds but of course sums up at some point...

File: docs/docs/30-administration/22-backends/40-kubernetes.md (hunk 1 above)
Comment (6543) [reply docs/docs/30-administration/22-backends/40-kubernetes.md:8]: Thanks :)

File: docs/docs/30-administration/00-setup.md (hunk 2 above)
Comment (anbraten) [reply docs/docs/30-administration/00-setup.md:177]: I would leave the top-level one as well, but add another one like this for the values file:

```suggestion
Have a look at the [`values.yaml`](https://github.com/woodpecker-ci/helm/blob/main/values.yaml) config files for all available settings.
```

**Comment by pat-s** [comment]
We need an approval :)

**Review by 6543 — APPROVED** [review]

**Pull request merged**